DetectorDescription/DDCMS/plugins/DDCutTubsFromPoints.cc

```diff
@@ -0,0 +1,194 @@
+  // slopes in r-Z, and the corner at r_max could stick out of the bounding
+  // volume otherwise.
+  double r = r_max;
+
+  // min and max z for the placement in the end
+  double min_z = 1e9;
+  double max_z = -1e9;
+
+  // counter of actually produced segments (excluding skipped ones)
+  int segment = 0;
+
+  // the segments and their corresponding offset (absolute, as in the input)
+  std::vector<dd4hep::Solid> segments;
+  std::vector<double> offsets;
+
+  Section s1 = sections[0];
+
+  for (Section s2 : sections) {
+    if (s1.phi != s2.phi) {
+      segment++;
+      // produce segment s1-s2.
+      std::string segname(solidOutput + "_seg_" + std::to_string(segment));
+
+      double phi1 = s1.phi;
+      double phi2 = s2.phi;
+
+      // track the min/max to properly place&align later
+      if (s2.z_l < min_z)
+        min_z = s2.z_l;
+      if (s2.z_t > max_z)
+        max_z = s2.z_t;
+
+      double P1_z_l = s1.z_l;
+      double P1_z_t = s1.z_t;
+      double P2_z_l = s2.z_l;
+      double P2_z_t = s2.z_t;
+
+      double P1_x_t = cos(phi1) * r;
+      double P1_x_l = cos(phi1) * r;
+      double P1_y_t = sin(phi1) * r;
+      double P1_y_l = sin(phi1) * r;
+
+      double P2_x_t = cos(phi2) * r;
+      double P2_x_l = cos(phi2) * r;
+      double P2_y_t = sin(phi2) * r;
+      double P2_y_l = sin(phi2) * r;
+
+      // each cutting plane is defined by P1-3. P1-2 are corners of the
+      // segment, P3 is at r=0 with the "average" z to get a nice cut.
+      double P3_z_l = (P1_z_l + P2_z_l) / 2;
+      double P3_z_t = (P1_z_t + P2_z_t) / 2;
+
+      // we only have one dz to position both planes. The anchor is implicitly
+      // between the P3's, we have to use an offset later to make the segments
+      // line up correctly.
+      double dz = (P3_z_t - P3_z_l) / 2;
+      double offset = (P3_z_t + P3_z_l) / 2;
+
+      // the plane is defined by P1-P3 and P2-P3; since P3 is at r=0 we
+      // only need the z.
+      double D1_z_l = P1_z_l - P3_z_l;
+      double D2_z_l = P2_z_l - P3_z_l;
+
+      // the normal is then the cross product...
+      double n_x_l = (P1_y_l * D2_z_l) - (D1_z_l * P2_y_l);
+      double n_y_l = (D1_z_l * P2_x_l) - (P1_x_l * D2_z_l);
+      double n_z_l = (P1_x_l * P2_y_l) - (P1_y_l * P2_x_l);
+
+      // ... normalized.
+      // flip the sign here (but not for t) since root wants it like that.
+      double norm = -sqrt(n_x_l * n_x_l + n_y_l * n_y_l + n_z_l * n_z_l);
+      n_x_l /= norm;
+      n_y_l /= norm;
+      n_z_l /= norm;
+
+      // same game for the t side.
+      double D1_z_t = P1_z_t - P3_z_t;
+      double D2_z_t = P2_z_t - P3_z_t;
+
+      double n_x_t = (P1_y_t * D2_z_t) - (D1_z_t * P2_y_t);
+      double n_y_t = (D1_z_t * P2_x_t) - (P1_x_t * D2_z_t);
+      double n_z_t = (P1_x_t * P2_y_t) - (P1_y_t * P2_x_t);
+
+      norm = sqrt(n_x_t * n_x_t + n_y_t * n_y_t + n_z_t * n_z_t);
+      n_x_t /= norm;
+      n_y_t /= norm;
+      n_z_t /= norm;
+
```

Comment:
A check or assert for `norm != 0.` should probably be added before dividing by `norm` here and after line 128.

Reply:
@cvuosalo As of now the main goal is to reproduce the same results with the migration. But additional checks do not hurt so I'll include them :+1: